x/cronos/keeper/evmhooks.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	gravitytypes "github.com/peggyjv/gravity-bridge/module/x/gravity/types"
+
+	"github.com/crypto-org-chain/cronos/x/cronos/types"
+)
+
+var (
+	// BankSendEvent represent the signature of
+	// `event __CosmosNativeBankSend(address recipient, uint256 amount, string denom)`
+	BankSendEvent abi.Event
+
+	// GravitySendEvent represent the signature of
+	// `event __CosmosNativeGravitySend(address recipient, uint256 amount, string denom)`
+	GravitySendEvent abi.Event
+)
+
+func init() {
+	addressType, _ := abi.NewType("address", "", nil)
+	uint256Type, _ := abi.NewType("uint256", "", nil)
+	stringType, _ := abi.NewType("string", "", nil)
+	BankSendEvent = abi.NewEvent(
+		"__CosmosNativeBankSend",
+		"__CosmosNativeBankSend",
+		false,
+		abi.Arguments{abi.Argument{
+			Name:    "recipient",
+			Type:    addressType,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "amount",
+			Type:    uint256Type,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "denom",
+			Type:    stringType,
+			Indexed: false,
+		}},
+	)
+	GravitySendEvent = abi.NewEvent(
+		"__CosmosNativeGravitySend",
```

Comment:
I think it would be better to define more generic event for solidity users that does not know much about cosmos ecosystem. The term "CosmosNative" is quite confusing

Something like "TransferToEthereum" or "SendToChain(chainName,...)"

Reply:
One thing to consider here is to try to prevent name conflict when some contract does not intent to call native functionality but defines identical event name and signatures.

Reply:
yeah maybe we can add a special character or prefix at the beginning to define the list of special instruction that can be uses in smart contract like 

@BankSend()  or CronosBankSend()
@IBCTransfer() or CronosIBCTransfer()
@EthereumTransfer() or CronosEthereumTransfer()

(Not sure if special character is allowed in events)

Reply:
- `__CronosNativeTransfer`
- `__CronosIBCTransfer`
- `__CronosEthereumTransfer`

Let's do this then.